pkg/management/component.go

```diff
@@ -116,20 +117,22 @@ type Component interface {
 
 // Component is the management component.
 type component struct {
-	keycloakClient KeycloakClient
-	eventDBModule  database.EventsDBModule
-	configDBModule ConfigurationDBModule
-	logger         keycloakb.Logger
+	keycloakClient          KeycloakClient
+	eventDBModule           database.EventsDBModule
+	configDBModule          ConfigurationDBModule
+	authorizedTrustIDGroups []string
```

Comment:
When instanciating this struct, you could convert the `[]string` into `map[string]bool`...
In SetTrustIDGroups, you would only have to check like this : `if _, ok := c.authorizedTrustIDGroups[groupName]; ok {...}` and avoid to have to check all values of your array (it would also make SetTrustIDGroups be shorter and so, easier to read)

Reply:
Done